test/test-parse-function.cpp

```diff
@@ -546,10 +546,15 @@ TEST(test_parse, function_statement_without_parameter_list_or_body) {
                                       "visit_variable_use",          // y
                                       "visit_variable_assignment",   // x
                                       "visit_end_of_module"));
-    EXPECT_THAT(v.errors,
-                ElementsAre(ERROR_TYPE_FIELD(
-                    error_missing_function_parameter_list, function_name,
-                    offsets_matcher(&code, strlen(u8"{ function "), u8"f"))));
+    EXPECT_THAT(
+        v.errors,
+        ElementsAre(
+            ERROR_TYPE_FIELD(
+                error_missing_function_parameter_list, function_name,
+                offsets_matcher(&code, strlen(u8"{ function "), u8"f")),
+            ERROR_TYPE_FIELD(
+                error_missing_function_body, function_name,
+                offsets_matcher(&code, strlen(u8"{ function "), u8"f"))));
```

Comment:
Minor: I think this test should remain as-is. One error seems better than two errors for this case IMO.

Idea: Make `parse_and_visit_function_parameters_no_scope` return a boolean. If the parameter list is present, it returns `true`; if the parameter list is missing, it returns `false`. In `parse_and_visit_function_parameters_and_body_no_scope`, stop parsing if `parse_and_visit_function_parameters_no_scope` returns `false`.